fe/fe-core/src/main/java/com/starrocks/qe/ConnectScheduler.java

```diff
@@ -142,6 +148,30 @@ public void unregisterConnection(ConnectContext ctx) {
                 conns.decrementAndGet();
             }
         }
+
+        if (userConnMetricsCounter.containsKey(ctx.getQualifiedUser())) {
```

Comment:
Could also use computeIfAbsent to avoid race contention.
```java
        userConnMetricsCounter.computeIfPresent(ctx.getQualifiedUser(), (k, v) -> {
            if (userConnMetricsCounter.get(ctx.getQualifiedUser()).getValue() <= 1) {
                MetricRepo.removeXXX();
                return null;
            }
            return v;
        });
```